### PR TITLE
Fix cluster creation logic for existing sessions

### PR DESCRIPTION
**Pull request by ricochet**
- Fix bug where multiple clusters could be created for a session
  With this logic, we can manually terminate clusters
  and vapor will self-heal when we try to create a new session.
- Inject s2apikey in config
- Add tests for:
  - test for primary use-case: new session, new cluster
  - calling /create (twice) on a session that has a cluster whose status is pending creation.
  - test for expired cluster.

### CODE REVIEW

File: pkg/routes.go [hunk 1]
```diff
@@ -116,26 +134,57 @@ func (s *server) handleCreate() http.HandlerFunc {
...
 			return
-		} else if errors.Is(err, sql.ErrNoRows) {
-			// no cluster found, create a new cluster
-			if err := s.createCluster(session, releaseId); err != nil {
+		}
+
+		if clusterId != "" {
+			// cluster already exists, check status
+			state, err := s.getClusterState(session, clusterId)
+			if err != nil && !errors.Is(err, sql.ErrNoRows) {
```
[the remark this answers is not in the document]
Comment (ricochet) [reply pkg/routes.go:147]: ErrNoRows was a possible state before we checked if we had a clusterId. Now that we do, it's not possible. Removed this.

File: pkg/sessions.go [hunk 2]
```diff
@@ -260,3 +225,7 @@ func (s *server) createDbUser(sessionId, clusterId string) error {
 
 	return s.db.SetSessionPassword(sessionId, sessionPassword)
 }
+
+func ClusterStateEnded(state mgmnt.ClusterState) bool {
+	return state == mgmnt.ClusterStateTerminated || state == mgmnt.ClusterStateSuspended || state == mgmnt.ClusterStateSuspending || state == mgmnt.ClusterStateFailed
+}
```
[the remark this answers is not in the document]
Comment (ricochet) [reply pkg/sessions.go:231]: Good catch

File: pkg/sessions_test.go [hunk 3]
```diff
@@ -244,17 +271,17 @@ func TestClusterStateEnded(t *testing.T) {
 		state    mgmnt.ClusterState
```
Comment (carlsverre) [inline pkg/sessions_test.go:271]: NIT: rename test

**Review by carlsverre — APPROVED** [review]

**Comment by carlsverre** [comment]
shipit

**Pull request merged**